demos/imagenet_classification_demo/main.cpp

```diff
@@ -0,0 +1,511 @@
+                lock.unlock();
+                emptyInferRequests.push({completedInferRequestInfo.inferRequest, 
+                                         std::vector<InferRequestInfo::InferRequestImage>()});
+
+                std::vector<unsigned> rightClasses = {};
+                for (size_t i = 0; i < completedInferRequestInfo.images.size(); i++) {
+                    rightClasses.push_back(completedInferRequestInfo.images[i].rightClass);
+                }
+
+                std::vector<std::vector<unsigned>> results = topResults(
+                    *completedInferRequestInfo.inferRequest.GetBlob(outputName), FLAGS_nt);
+                std::vector<std::string> predictedLabels = {};
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    int predictionResult = -1;
+                    if (!FLAGS_gt.empty()) {
+                        for (size_t j = 0; j < FLAGS_nt; j++) {
+                            unsigned predictedClass = results[i][j];
+                            if (predictedClass == rightClasses[i]) {
+                                predictionResult = 1;
+                                predictedLabels.push_back(labels[predictedClass]);
+                                correctPredictionsCount++;
+                                break;
+                            }
+                        }
+                    } else {
+                        predictionResult = 0;
+                    }
+
+                    if (predictionResult != 1) {
+                        predictedLabels.push_back(labels[results[i][0]]);
+                    }
+                    totalPredictionsCount++;
+
+                    shownImagesInfo.push_back(
+                        std::make_tuple(completedInferRequestInfo.images[i].mat, predictedLabels[i], predictionResult));
+                }
+
+                framesNum += FLAGS_b;
+
+                avgFPS = 1. * framesNum / std::chrono::duration_cast<std::chrono::milliseconds>(
+                    std::chrono::steady_clock::now() - startTickCount).count() * 1000;
+                
+                gridMat.updateMat(shownImagesInfo);
+                auto processingEndTime = std::chrono::steady_clock::now();
+                if (!FLAGS_no_show) {
+                    cv::imshow("imagenet_classification_demo", gridMat.outImg);
+                    key = static_cast<char>(cv::waitKey(1));
+                }
+
+                for (size_t i = 0; i < FLAGS_b; i++) {
+                    latencySum += (1. * std::chrono::duration_cast<std::chrono::milliseconds>(
+                        processingEndTime - completedInferRequestInfo.images[i].startTime).count() / 1000);
+                }
+                avgLatency = latencySum / framesNum;
+                accuracy = static_cast<double>(correctPredictionsCount) / totalPredictionsCount;
+                gridMat.textUpdate(avgFPS, avgLatency, accuracy, isTestMode, !FLAGS_gt.empty(), presenter);
```

Comment:
This part should go before `imshow`: that allows to report the latest results

Reply:
This `for` can be rewritten as a for range based loop. This removes an assumption, that `completedInferRequestInfo.images.size()` is `FLAGS_b`, even though it's true. Latency for all images is to be computed, but not only for the first `FLAGS_b`.